addons/mrp/models/mrp_workorder.py

```diff
@@ -325,6 +325,9 @@ def _compute_duration_expected(self):
         for workorder in self:
             if workorder.state not in ['done', 'cancel']:
                 workorder.duration_expected = workorder._get_duration_expected()
+                # The date_finished should be recomputed if the duration_expected is changed
+                if workorder.date_start:
```

Comment:
@Whenrow How would you do this ? Update the `duration_expected` and `date_finished` in the write when we are writing a `workcenter_id` ?

Reply:
![image](https://github.com/odoo/odoo/assets/32939472/87e09fa6-5197-4adb-b233-f31765cd66d8)
Something like this is working but it's doing the exact same as the compute

Reply:
@robinengels I prefer doing this because you test what field triggers the change as well as the status of the workorder. In the `_compute` we don't know this information.